cmsdev/internal/test/ims/verify_ims_recipes.go

```diff
@@ -34,28 +35,57 @@ import (
  *
  */
 
-func TestRecipeCRUDOperation() (passed bool) {
+func TestRecipeCRUDOperationUsingAPIVersions() (passed bool) {
+	passed = true
+	for _, apiVersion := range common.IMSAPIVERSIONS {
+		common.PrintLog(fmt.Sprintf("Testing recipe CRUD operations using IMS API version: %s", apiVersion))
+		common.SetIMSAPIVersion(apiVersion)
+		passed = passed && TestRecipeCRUDOperation(apiVersion)
+	}
+
+	// Reset the IMS API version to the default value
+	common.SetIMSAPIVersion("")
+	common.PrintLog("Testing recipe CRUD operations using default API version")
+	passed = passed && TestRecipeCRUDOperation(common.GetIMSAPIVersion())
+	return passed
+}
+
+func TestRecipeCRUDOperation(apiVersion string) (passed bool) {
 	// Test creating a recipe
 	recipeRecord, success := TestRecipeCreate()
 	if !success {
 		return false
 	}
-	// Test updating the recipe
-	updated := TestRecipeUpdate(recipeRecord.Id)
 
-	// Test soft deleting the recipe
-	deleted := TestRecipeDelete(recipeRecord.Id)
+	if apiVersion == "v3" {
+		// Test updating the recipe
+		updated := TestRecipeUpdate(recipeRecord.Id)
```

Comment:
Like with the images case, seems like this could go outside of the conditional, since it's common to both branches